ethereum/dataproviders/src/test/java/tech/pegasys/teku/dataproviders/generators/CachingTaskQueueTest.java

```diff
@@ -276,10 +303,29 @@ private void assertRebasedTaskCount(final int expectedCount) {
   }
 
   private void assertPendingTaskCount(final int expectedCount) {
-    final double value =
+    final OptionalDouble optionalValue =
         metricsSystem
-            .getGauge(TekuMetricCategory.STORAGE, METRICS_PREFIX + "_tasks_requested")
-            .getValue();
+            .getLabelledGauge(TekuMetricCategory.STORAGE, METRICS_PREFIX + "_tasks")
+            .getValue("requested");
+    final double value = optionalValue.orElseThrow();
+    assertThat(value).isEqualTo(expectedCount);
```

Comment:
nit: We can simplify this a bit: 

```suggestion
    assertThat(optionalValue).hasValue(expectedCount);
```

and do the same for the other `OptionalDouble` asserts below.